packages/one-app-dev-bundler/esbuild/plugins/externals-loader.js

```diff
@@ -40,20 +74,74 @@ const externalsLoader = ({ bundleType }) => ({
     // return a namespace. (see this above)
     // your onLoad can then just match .* within that namespace and you guarantee you target
     // every package you want.
-    build.onLoad({ filter: /.*/, namespace: 'externalsLoader' }, async (args) => {
-      const jsContent = `\
-try {
-  module.exports = ${globalReferenceString}.getTenantRootModule().appConfig.providedExternals['${args.path}'].module;
-} catch (error) {
-  const errorGettingExternal = new Error('Failed to get external ${args.path} from root module');
-  errorGettingExternal.shouldBlockModuleReload = false;
-  throw errorGettingExternal;
-}
-`;
+    build.onLoad({ filter: /.*/, namespace: 'externalsLoader' }, async ({ path: externalName, ...args }) => {
+      if (bundleType === BUNDLE_TYPES.SERVER) {
+        const require = createRequire(import.meta.url);
+        const cjsCode = (await swc.transform(await fs.promises.readFile(require.resolve(externalName), 'utf8'), {
```

Comment:
Why use swc here? this is an esbuild based bundler?

Reply:
the reason we use `swc` here is because the external's code could be ESM. Without transpilation, the ESM code will be injected. This is only for the nodejs fallback.

Reply:
Can esbuild not do that?

Reply:
yes and this will be replaced with the new server external fallbacks implementation